flashlight/fl/tensor/backend/jit/eval/Evaluator.cpp

```diff
@@ -52,16 +52,45 @@ void Evaluator::evalCustomNode(CustomNode& node) {
 }
 
 void Evaluator::evalIndexNode(IndexNode& node) {
-  std::vector<Index> indices;
-  for (const auto& index : node.indices()) {
+  const auto& indexedTensor = node.indexedNode()->getResult().value();
+  node.setResult(indexedTensor(unwrapTensorInIndices(node.indices())));
+}
+
+void Evaluator::evalIndexedUpdateNode(IndexedUpdateNode& node) {
+  // TODO no need to copy if indexedNode has only 1 user here
+  auto indexedTensor = node.indexedNode()->getResult().value().copy();
+  const auto firstUnwrappedIndices =
+      unwrapTensorInIndices(node.indexings().front());
+  const auto& updateDataTensor = node.updateDataNode()->getResult().value();
+  // if we do X = Y, it's a copy instead of update, thus the special case here
```

Comment:
I think I'm not understanding the comment -- why would assignment result in an IndexedUpdateNode?